### ISSUE TITLE
Question about the implementation of contrast loss

### ISSUE THREAD

**Issue by Sherrylife** [blackfeather-wang/InfoPro-Pytorch#4]
Hi, @blackfeather-wang, I would like to ask if there is any discrepancy between your code implementation and the description of the paper. In your paper, you say that we are going to minimize $\mathcal{L}_{contrast}$, and the form of $\mathcal{L}_{contrast}$ is as follows.
![image](https://user-images.githubusercontent.com/60345931/236372410-0925ea2d-cf62-4350-a037-b4fcc04e5567.png)

However, after I read your code, I find that your code seems to minimize $-\mathcal{L}_{contrast}$, and the consequence is that you're not maximizing the lower bound of $I(h,y)$, instead you're minimizing the lower bound of $I(h,y)$, can you explain that?

**Comment by blackfeather-wang** [comment]
Thank you for the comment. Indeed, there is a small typo in Eq. (4) of the paper: log() should be -log(). Note that only -log() can contribute to a meaningful form of loss function. Minimizing log() makes no sense. All the proof and the implementation is based on -log(). Everything will be correct after fixing this typo.

**Comment by Sherrylife** [comment]
> Thank you for the comment. Indeed, there is a small typo in Eq. (4) of the paper: log() should be -log(). Note that only -log() can contribute to a meaningful form of loss function. Minimizing log() makes no sense. All the proof and the implementation is based on -log(). Everything will be correct after fixing this typo.

Thank you for your reply. Does this mean that the derivation of Appendix D is correct?

**Comment by blackfeather-wang** [comment]
Yes. The only issue is the typo in Eq. (4), i.e.,  log() should be -log(). Appendix D considers -log() as well.

**Comment by Sherrylife** [comment]
Thank you for your patient reply.
